nativelib/src/main/resources/dirent.c

```diff
@@ -0,0 +1,39 @@
+#include <dirent.h>
+#include <string.h>
+#include <gc.h>
+
+#define NAME_MAX 255
+
+struct scalanative_dirent {
+    unsigned long long d_ino;  /** file serial number */
+    char d_name[NAME_MAX + 1]; /** name of entry */
+};
+
+DIR *scalanative_opendir(const char *name) {
+
+    return opendir(name);
+}
+
+void scalanative_dirent_init(struct dirent *dirent, struct scalanative_dirent *my_dirent) {
+    my_dirent->d_ino  = dirent->d_ino;
+    strncpy(my_dirent->d_name, dirent->d_name, NAME_MAX);
+    my_dirent->d_name[NAME_MAX] = '\0';
+}
+
+struct scalanative_dirent *scalanative_readdir(DIR *dirp) {
+    struct dirent *buf = readdir(dirp);
+    if (buf != NULL) {
+        struct scalanative_dirent *my_dirent =
+            (struct scalanative_dirent *) GC_malloc(sizeof(struct scalanative_dirent));
```

Comment:
Same comment as for `scalanative_stat_copy` applies here. Need to take an extra argument, instead of GC allocating the result.